test/model/message_test.dart

```diff
@@ -123,6 +127,269 @@ void main() {
     });
   });
 
+  group('edit-message methods', () {
+    late StreamMessage message;
+    Future<void> prepareEditMessage() async {
+      await prepare();
+      message = eg.streamMessage();
+      await prepareMessages([message]);
+      check(connection.takeRequests()).length.equals(1); // message-list fetchInitial
+    }
+
+    void checkRequest(int messageId, String content) {
+      check(connection.takeRequests()).single.isA<http.Request>()
+        ..method.equals('PATCH')
+        ..url.path.equals('/api/v1/messages/$messageId')
+        ..bodyFields.deepEquals({
+          'content': content,
+        });
+    }
+
+    test('smoke', () => awaitFakeAsync((async) async {
+      await prepareEditMessage();
+      check(store.getEditMessageErrorStatus(message.id)).isNull();
+
+      connection.prepare(
+        json: UpdateMessageResult().toJson(), delay: Duration(seconds: 1));
+      store.editMessage(messageId: message.id, content: 'new content');
+      checkRequest(message.id, 'new content');
+      checkNotifiedOnce();
+
+      async.elapse(Duration(milliseconds: 500));
+      // Mid-request
+      check(store.getEditMessageErrorStatus(message.id)).isNotNull().isFalse();
+
+      async.elapse(Duration(milliseconds: 500));
+      // Request has succeeded; event hasn't arrived
+      check(store.getEditMessageErrorStatus(message.id)).isNotNull().isFalse();
+
+      await store.handleEvent(eg.updateMessageEditEvent(message));
+      check(store.getEditMessageErrorStatus(message.id)).isNull();
+      checkNotifiedOnce();
+    }));
+
+    test('concurrent edits on different messages', () => awaitFakeAsync((async) async {
+      await prepareEditMessage();
+      final otherMessage = eg.streamMessage();
+      await store.addMessage(otherMessage);
+      checkNotifiedOnce();
+
+      check(store.getEditMessageErrorStatus(message.id)).isNull();
+
+      connection.prepare(
+        json: UpdateMessageResult().toJson(), delay: Duration(seconds: 1));
+      store.editMessage(messageId: message.id, content: 'new content');
+      checkRequest(message.id, 'new content');
+      checkNotifiedOnce();
+
+      async.elapse(Duration(milliseconds: 500));
+      // Mid-first request
+      check(store.getEditMessageErrorStatus(message.id)).isNotNull().isFalse();
+      check(store.getEditMessageErrorStatus(otherMessage.id)).isNull();
+      connection.prepare(
+        json: UpdateMessageResult().toJson(), delay: Duration(seconds: 1));
+      store.editMessage(messageId: otherMessage.id, content: 'other message new content');
+      checkRequest(otherMessage.id, 'other message new content');
+      checkNotifiedOnce();
+
+      async.elapse(Duration(milliseconds: 500));
+      // First request has succeeded; event hasn't arrived
+      // Mid-second request
+      check(store.getEditMessageErrorStatus(message.id)).isNotNull().isFalse();
+      check(store.getEditMessageErrorStatus(otherMessage.id)).isNotNull().isFalse();
```

Comment:
How about having `checkNotNotified` calls here and later? So that we are extra sure that it's the edit event that triggers the notifications.